packages/@aws-cdk/aws-cognito/lib/user-pool-idps/saml.ts

```diff
@@ -0,0 +1,103 @@
+import { Names, Token } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { CfnUserPoolIdentityProvider } from '../cognito.generated';
+import { UserPoolIdentityProviderProps } from './base';
+import { UserPoolIdentityProviderBase } from './private/user-pool-idp-base';
+
+/**
+ * Properties to initialize UserPoolIdentityProviderSaml.
+ */
+export interface UserPoolIdentityProviderSamlProps extends UserPoolIdentityProviderProps {
+  /**
+   * The name of the provider.
+   *
+   * @default - the unique ID of the construct
+   */
+  readonly name?: string;
+
+  /**
+   * Identifiers
+   *
+   * Identifiers can be used to redirect users to the correct IdP in multitenant apps.
+   *
+   * @default - no identifiers used
+   */
+  readonly identifiers?: string[]
+
+  /**
+   * The SAML metadata file content.
+   *
+   * @default - no file content specified
+   */
+  readonly metadataFile?: string;
+
+  /**
+   * The SAML metadata URL.
+   *
+   * @default - no URL specified
+   */
+  readonly metadataUrl?: string;
```

Comment:
Instead of having these be mutually exclusive and optional, let's have just one field that is required that can be of either type that we translate into the right field.

Reply:
Sounds good -- I replaced these with `metadataType` and `metadataContent` props, both required. The former is an enum that determines how the latter should be interpreted (as a URL, or as XML content). This removes the need for the mutually-exclusive validation. Is that what you had in mind?

Reply:
What I meant here was more along the lines of how we handle the different types of input that we turn into [Schedule](https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/aws-events/lib/schedule.ts). I think that same conceptual approach can be used here. My apologies for not providing further context and that link here in the first place.

Reply:
I see! Trying to follow the example of `Schedule`, I've reworked it now to just have one field of type `UserPoolIdentityProviderSamlMetadata`, which can only be created by one of two methods, `url` and `file`. Using the appropriate one should ensure that the `providerDetails` in the CloudFormation resource are set correctly (translated to `MetadataURL` and `MetadataFile` respectively).